src/ado/lbl_use_meta.ado

```diff
@@ -0,0 +1,102 @@
+*! version XX XXXXXXXXX ADAUTHORNAME ADCONTACTINFO
+
+cap program drop   lbl_use_meta
+    program define lbl_use_meta, rclass
+
+qui {
+
+  version 14.1
+
+  * Update the syntax. This is only a placeholder to make the command run
+  syntax, ///
+  Varlist(varlist)    ///
+  FROM_char(string)   ///
+  [                   ///
+    TEMplate(string)  ///
+    APPly_to(string)  ///
+    MISSing_ok    ///
+  ]
+
+  * Create a dummy that is one if multiple vars was passed in varlist()
+  local multi_varlist = (`: list sizeof varlist' > 1)
+
+  ****************************
+  * Check char on varlist
+
+  * Test which variables that has the char
+  ds `varlist', has(char `from_char')
+  local varlist_with_char "`r(varlist)'"
+
+  * List variables in varlist that did not have the char
+  local varlist_without_char : list varlist - varlist_with_char
+  * Out put a warning for variables that did not have
+  if !missing("`varlist_without_char'") {
+    noi di as text "{pstd}{red: Warning}: No action will be taken on these variables included in {inp:varlist()} as they did not have the char {inp:`from_char'}:{p_end}"
+    noi di as text "{phang}- `varlist_without_char'{p_end}"
+  }
+
+
+  * Test that at least one variable had the char
+  if (missing("`varlist_with_char'") & missing("`missing_ok'")) {
+    noi di as error "{pstd}No variable in {inp:varlist()} had the char {inp:`from_char'}.{p_end}"
+    error 99
+    exit
+  }
+
+  ****************************
+  * Exectue command on single-var varlist
+
+  ****************************
+  * Use meta char for each var
+
+  foreach var of local varlist_with_char {
+    * Get the char
+    local value : char `var'[`from_char']
+    * Call returned value different if one or several var in varlist
+    if (`multi_varlist' == 0) return local char_value `"`value'"'
+    else return local c_`var' `"`value'"'
+
+    * If template is used apply it
+    if !missing(`"`template'"') {
+      * Populate the template
+      apply_template , template(`"`template'"') value(`"`value'"')
+      *Overwrite value with template
+      local value `"`r(template)'"'
+
+      * Call returned value different if one or several var in varlist
+      if (`multi_varlist' == 0) return local template `"`value'"'
+      else return local t_`var' `"`value'"'
+    }
+
+    ****************************
+    * If apply_to is used, then apply the meta value
+
+    if ("`apply_to'" == "varlabel") {
+      label variable `var' `"`value'"'
+    }
+    // Add more apply_to here if applicable in future
+  }
+
+  if (`multi_varlist' != 0) {
+    return local varlist_with_char    "`varlist_with_char'"
+    return local varlist_without_char "`varlist_without_char'"
+  }
+}
+end
+
+cap program drop   apply_template
+    program define apply_template, rclass
+
+  syntax , template(string) value(string)
+
+  if (strpos(`"`template'"', "{META}") == 0) {
```

Comment:
In the jargon of the command, we seem to use `char` and metadata somewhat interchangeably. Would it be good to stick with one? If so, would it make sense to use `char`? And if we decide in that direction, shouldn't we use `{CHAR}` as the placeholder?

If we anticipate this command being directly used by the LSMS team, I might favor "metadata", even if that remains a bit ambiguous. If we anticipate this command being used by a broader audience, then "char" might be preferred, since it has a clear definition and referenced implementation in Stata. (The LSMS team could get educated on chars by this command and the selector package calling their attention to it.)

This isn't a technical problem, but instead one about terminology, and consistency in usage of that terminology. I'm personally open to either solution.

Reply:
You are raising a good point. 

I went with `char` to make it clear that this command can be used with char regardless if that char is meta data from `sel_add_metadata` or from anywhere else. 

But I think the most common user, at least initially, will be the LSMS team, so I am happy to go with what you think is best for them. 

And a `char` is a type of meta data so I do not think meta data is a bad way to refer to a char.

Reply:
I have updated the name of the option `from_char` to `from_meta`. This looks much better. I updated the text file and documentation as well. 

But I am sure the documentation can benefit from your review. So feel free to do that. I am more literate in Stata than in English :)

Reply:
> I am more literate in Stata than in English :)

You sell yourself far too short. Your (technical) English is much better than mine.

I rewrote a few sections of documentation for this command slightly with the end-user in mind. I'm not convinced I got it write.

Feel free to revise--or simply to merge and have us revise later.

Reply:
This looks really good. 

I have rendered the file and I have also in `ad_sthlp` fixes so a line can end in `` ` ``.